UnityProject/Assets/Scripts/Objects/Closets/ClosetControl.cs

```diff
@@ -245,23 +245,33 @@ public void ReleaseObjects()
 			objectContainer.RetrieveObjects();
 		}
 
-		public void EntityTryEscape(GameObject performer)
+		public void EntityTryEscape(GameObject performer, Action ifCompleted)
 		{
 			// First, try to just open the closet. Anything can do this.
 			if (IsLocked == false && IsWelded == false)
 			{
 				SetDoor(Door.Opened);
+				ifCompleted?.Invoke();
 				return;
 			}
 
+			GameObject sourceobjbehavior = gameObject;
+			RegisterObject sourceregisterobject = registerObject;
+			if (GetComponent<ObjectBehaviour>().parentContainer != null)
+			{
+				sourceobjbehavior = GetComponent<ObjectBehaviour>().parentContainer.gameObject;
```

Comment:
You can use the cached reference assigned to `pushPull`. We're pretty big on caching component references.